### PR TITLE
PRIME-2270 New HA Site Flow

### PR DESCRIPTION
**Pull request by bergomi02**
- The new flow: Care Type -> Vendor -> Site Detail (remove Security Group field)

https://pr-2201.pharmanetenrolment.gov.bc.ca/health-authority
https://pr-2201.pharmanetenrolment.gov.bc.ca/admin

### CODE REVIEW

**Review by Paahn — APPROVED** [review]
LGTM

File: prime-dotnet-webapi/Controllers/HealthAuthoritySitesController.cs [hunk 1]
```diff
@@ -60,9 +60,9 @@ public async Task<ActionResult> CreateHealthAuthoritySite(int healthAuthorityId,
             {
                 return Forbid();
             }
-            if (!await _healthAuthorityService.VendorExistsOnHealthAuthorityAsync(healthAuthorityId, payload.HealthAuthorityVendorId))
+            if (!await _healthAuthorityService.VendorExistsOnHealthAuthorityAsync(healthAuthorityId, payload.HealthAuthorityCareTypeId))
```
Comment (neophyte57) [inline prime-dotnet-webapi/Controllers/HealthAuthoritySitesController.cs:63]: This seems to be calling the wrong method, that is, checking vendors with care type ID?

Comment (bergomi02) [reply prime-dotnet-webapi/Controllers/HealthAuthoritySitesController.cs]: Good catch. Not sure what I was thinking. The validation has been moved to Update/PUT mothed and revert back to passing VendorId.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-site.model.ts [hunk 2]
```diff
@@ -89,17 +89,17 @@ export class HealthAuthoritySite extends AbstractBaseHealthAuthoritySite impleme
...
    */
   public forCreate(authorizedUserId: number): HealthAuthoritySiteCreate {
-    const healthAuthorityVendorId = this.healthAuthorityVendor?.id;
+    const healthAuthorityCareTypeId = this.healthAuthorityCareType?.id;
     if (!authorizedUserId) {
       throw Error('Authorized user identifier was not provided');
     }
-    if (!healthAuthorityVendorId) {
+    if (!healthAuthorityCareTypeId) {
       throw Error('Health authority vendor identifier was not provided');
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-site.model.ts:97]: Error message needs updating

Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-site.model.ts]: fixed

**Comment by neophyte57** [comment]
I think that the non-UI portions associated with "Security Group" should be removed too (e.g. `site-information-form-state.class.ts`, `HealthAuthoritySite.cs`) but probably in a separate ticket

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts [hunk 3]
```diff
@@ -63,9 +69,9 @@ export class HealthAuthCareTypePageComponent extends AbstractHealthAuthoritySite
 
   protected patchForm(): void {
     const healthAuthId = +this.route.snapshot.params.haid;
-    const healthAuthSiteId = +this.route.snapshot.params.sid;
-    if (!healthAuthId || !healthAuthSiteId) {
-      throw new Error('No health authority site ID was provided');
+    if (!healthAuthId) {
+      // Don't throw an error as new registrations are created in this view
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts:73]: Shouldn't there still be an error if no HA ID?

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts [hunk 4]
```diff
@@ -74,10 +80,37 @@ export class HealthAuthCareTypePageComponent extends AbstractHealthAuthoritySite
     this.healthAuthoritySiteFormStateService.setForm(site, !this.hasBeenSubmitted);
   }
 
-  protected afterSubmitIsSuccessful(): void {
+  protected submissionRequest(): Observable<unknown> {
+    const { haid, sid } = this.route.snapshot.params;
+    const healthAuthoritySite = this.healthAuthoritySiteFormStateService.json;
+    console.log("testing");
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts:86]: Debugging left-over

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts (hunk 3 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts:73]: It is transferred from vendor page, which was the starting page, to the new starting page.
image
I was thinking to inherit the existing behavior by doing so.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts (hunk 4 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts:86]: removed

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts (hunk 3 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/health-auth-care-type-page/health-auth-care-type-page.component.ts:73]: It is transferred from vendor page, which was the starting page, to the new starting page.
image

I was thinking to inherit the existing behavior by doing so.

**Review by neophyte57 — APPROVED** [review]
Looks good!  FYI, created this Jira issue for further clean-up:  https://bcgovmoh.atlassian.net/browse/PRIME-2336

**Pull request merged**